prqlc/prqlc-ast/src/expr/ident.rs

```diff
@@ -67,10 +67,18 @@ impl Ident {
             .all(|(prefix_component, self_component)| prefix_component == self_component)
     }
 
+    pub fn starts_with_path<S: AsRef<str>>(&self, prefix: &[S]) -> bool {
+        if prefix.len() > self.path.len() + 1 {
```

Comment:
I'm probably being slow, but why the `+ 1`? If the prefix is longer than the path, isn't that dispositive?

Reply:
Prefix is a slice, self is an Ident. Ident has lentgh of its path, plus one for the name field.

Looks like a bug, but I don't think it is.